internal/rootcoord/step.go

```diff
@@ -339,3 +341,36 @@ func (s *nullStep) Desc() string {
 func (s *nullStep) Weight() stepPriority {
 	return stepPriorityLow
 }
+
+type AlterCollectionStep struct {
+	baseStep
+	oldColl *model.Collection
+	newColl *model.Collection
+	ts      Timestamp
+}
+
+func (a *AlterCollectionStep) Execute(ctx context.Context) ([]nestedStep, error) {
+	err := a.core.meta.AlterCollection(ctx, a.oldColl, a.newColl, a.ts)
+	return nil, err
+}
+
+func (a *AlterCollectionStep) Desc() string {
+	return fmt.Sprintf("alter collection, collectionID: %d, ts: %d", a.oldColl.CollectionID, a.ts)
+}
+
+type BroadcastAlteredCollectionStep struct {
```

Comment:
I've seen any place cached collection schema, such as datanode segment_replica, querycoord.

Although that might be not related to TTL functions, should we think of mechainsum to cleanup cache when alter collection?